lib/swaps/Swaps.ts

```diff
@@ -669,16 +675,39 @@ class Swaps extends EventEmitter {
         return deal.rPreimage;
       } catch (err) {
         this.failDeal(deal, SwapFailureReason.SendPaymentFailure, err.message);
-        return 'Got exception from sendPaymentSync' + err.message;
+        throw new Error(`Got exception from sendPaymentSync ${err.message}`);
       }
     } else {
       // If we are here we are the taker
-      this.logger.debug('Executing taker code');
+      assert(deal.rPreimage, 'preimage must be known if we are the taker');
+      assert(htlcCurrency === undefined || htlcCurrency === deal.takerCurrency, 'incoming htlc does not match expected deal currency');
+      this.logger.debug('Executing taker code to resolve hash');
 
       this.setDealPhase(deal, SwapPhase.AmountReceived);
-      return deal.rPreimage;
+      return deal.rPreimage!;
+    }
+  }
+
+  public handleResolveRequest = async (resolveRequest: ResolveRequest): Promise<string> => {
+    const rHash = resolveRequest.getHash();
+    const amount = resolveRequest.getAmount() / 1000;
```

Comment:
We're dividing by 1000 here because of millisatoshis, right?

Reply:
Yup, although this code is probably outdated now and we should confirm the units that we're getting from raiden. But previously yes the resolve request was in millisatoshis.